Mailjet documentation (#943)
rudderlabs/rudderstack-docs: docs/destinations/streaming-destinations/mailjet.mdx

</div>

## Getting started

RudderStack supports sending events to Mailjet via the following <Link to="/destinations/rudderstack-connection-modes/">connection modes</Link>:

| **Connection Mode** | **Web**       | **Mobile**    | **Server**    |
| :------------------ | :------------ | :------------ | :------------ |
| **Device mode**     |       -        |        -       |        -       |
| **Cloud mode**      |      Supported       |     Supported    |    Supported     |

Once you have confirmed that the source platform supports sending events to Mailjet, follow these steps:

1. From your [RudderStack dashboard](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **Mailjet**.
2. Assign a name to your destination and click **Continue**.

### Connection settings 

To successfully set up Mailjet as a destination, you need to configure the following settings:

- **Mailjet API Key**: Enter your Mailjet API key. For more information on obtaining your API key, refer to the <Link to="#faq">FAQ</Link> section below.

> - **Mailjet API Secret**: Enter your Mailjet secret key.

Author: Please confirm if this should be the primary API key and secret or the **Sub-account key and secret** @mihir-4116 @shrouti1507

Reviewer: It should be apiKey and Secret

> - **Mailjet Contact List ID**: Enter your Mailjet contact list ID. RudderStack will use this ID if the list ID is not provided via the `externalId` field in the event's payload.

Author: @mihir-4116 @shrouti1507 how does one get the contact list ID? this information was not present in the spec.

Reviewer: When any user creates any contact list, the id will be shown in ui when we navigate to contact lists

| :------------------ | :------------ | :------------ | :------------ |
| **Device mode**     |       -        |        -       |        -       |
| **Cloud mode**      |      Supported       |     Supported    |    Supported     |

Once you have confirmed that the source platform supports sending events to Mailjet, follow these steps:

1. From your [RudderStack dashboard](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **Mailjet**.
2. Assign a name to your destination and click **Continue**.

### Connection settings 

To successfully set up Mailjet as a destination, you need to configure the following settings:

- **Mailjet API Key**: Enter your Mailjet API key. For more information on obtaining your API key, refer to the <Link to="#faq">FAQ</Link> section below.
- **Mailjet API Secret**: Enter your Mailjet secret key.
- **Mailjet Contact List ID**: Enter your Mailjet contact list ID. RudderStack will use this ID if the list ID is not provided via the `externalId` field in the event's payload. For more information on obtaining your contact list ID, refer to the <Link to="#faq">FAQ</Link> section below.

> - **Map RudderStack user attributes to Mailjet contact properties**: Use this setting to map your RudderStack event properties to specific custom properties created in the Mailjet dashboard. For more information on creating and managing contact properties, refer to this <a href="https://documentation.mailjet.com/hc/en-us/articles/360043176353-How-to-create-manage-contact-properties-">Mailjet documentation</a>.

Reviewer 2: ```suggestion
- **Map RudderStack user attributes to Mailjet contact properties**: Use this setting to map your RudderStack event properties to specific custom properties created in the Mailjet dashboard. For more information on creating and managing contact properties, refer to the <a href="https://documentation.mailjet.com/hc/en-us/articles/360043176353-How-to-create-manage-contact-properties-">Mailjet documentation</a>.
```

<div class="warningBlock">
Mailjet will not create a contact for the event properties configured in the RudderStack dashboard but not created in Mailjet.
</div>

## Identify

You can use the <Link to="/event-spec/standard-events/identify/">`identify`</Link> call to add or remove a contact in Mailjet. If a contact already exists, RudderStack updates the contact details.

A sample `identify` call is shown below:

```javascript
rudderanalytics.identify('1hKOmRA4el9Z', {
      firstName: 'Alex',
      lastName: 'Keener',
      email: "alex@example.com"
    }, {
      externalId: [{
        type: "listId",
        id: "13314el9Z"
      }],
      integrations: {
        mailjet: {
          Action: "remove"
        }
      }
    );
```

When sending events to Mailjet, note the following:

- The `email` field is mandatory. Otherwise, Mailjet rejects the events.

> - You can specify the Mailjet list ID either via the RudderStack dashboard or from the `externalId` field.

Reviewer 2: ```suggestion
- You can specify the Mailjet list ID either in the **Mailjet Contact List ID** setting of the RudderStack dashboard or in the `externalId` field.
```

> - You can specify the Mailjet action (adding/removing user) in the `Action` field of the `integrations` object. If no action is specified, RudderStack sets this field to `addnoforce`, by default.

Reviewer 2: Would be great if we could mention what `addnoforce` would mean..

Author: Added this info.

### Supported mappings

The following table lists the mappings between the RudderStack attributes and the Mailjet properties:

| RudderStack property | Mailjet property |
| :--------------| :---------|
| `traits.email` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `Email` |
| `traits.name` | `Name` |

> | `context.traits.IsExcludedFromCampaigns`, `traits.IsExcludedFromCampaigns` | `IsExcludedFromCampaigns` |

Reviewer 2: ```suggestion
| `context.traits.IsExcludedFromCampaigns` <br/> `traits.IsExcludedFromCampaigns` | `IsExcludedFromCampaigns` |
```
